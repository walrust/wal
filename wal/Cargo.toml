[package]
name = "wal"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
wal-component = {path = "../wal-component"}
wal-vdom = {path = "../wal-vdom"}

[dependencies.web-sys]
version = "0.3.64"
features = [
    "console", 
    "Window",
    "Document",
    "Node",
    "Element",
    "Text",
]