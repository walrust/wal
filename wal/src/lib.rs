pub mod virtual_dom;
pub mod component;
pub mod app;
pub mod utils;