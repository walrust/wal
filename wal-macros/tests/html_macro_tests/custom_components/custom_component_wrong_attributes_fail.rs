use wal::{
    component::{component_node::ComponentBehavior, Component},
    virtual_dom::VNode,
};
use wal_macros::html;

include!("../utils/custom_components/custom_component_props_i32.rs");

fn main() {
    html! { <CustomComponentPropsI32 attr = 5 /> };
    html! { <CustomComponentPropsI32 props = 1 attr = 5 /> };
    html! { <CustomComponentPropsI32 key = "key" attr = 5 /> };
    html! { <CustomComponentPropsI32 props = 1 attr = 5 key="key" /> };
    html! { <CustomComponentPropsI32 props = 0 props = 1 /> };
    html! { <CustomComponentPropsI32 props = 0 key = "key" props = 1 /> };
    html! { <CustomComponentPropsI32 key = "key1" key = "key2" /> };
    html! { <CustomComponentPropsI32 key = "key1" props = 0 key = "key2" /> };
    html! { <CustomComponentPropsI32 props /> };
    html! { <CustomComponentPropsI32 props = /> };
    html! { <CustomComponentPropsI32 props = key = "key" /> };
    html! { <CustomComponentPropsI32 key /> };
    html! { <CustomComponentPropsI32 key = /> };
    html! { <CustomComponentPropsI32 key = props = 1 /> };
}
