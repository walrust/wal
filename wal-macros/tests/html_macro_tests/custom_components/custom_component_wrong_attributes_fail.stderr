error: Unsupported attribute `attr`. Custom components supports only `props` and `key` attributes
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:10:38
   |
10 |     html! { <CustomComponentPropsI32 attr = 5 /> };
   |                                      ^^^^

error: Unsupported attribute `attr`. Custom components supports only `props` and `key` attributes
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:11:48
   |
11 |     html! { <CustomComponentPropsI32 props = 1 attr = 5 /> };
   |                                                ^^^^

error: Unsupported attribute `attr`. Custom components supports only `props` and `key` attributes
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:12:50
   |
12 |     html! { <CustomComponentPropsI32 key = "key" attr = 5 /> };
   |                                                  ^^^^

error: Unsupported attribute `attr`. Custom components supports only `props` and `key` attributes
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:13:48
   |
13 |     html! { <CustomComponentPropsI32 props = 1 attr = 5 key="key" /> };
   |                                                ^^^^

error: Duplicate attribute `props`
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:14:48
   |
14 |     html! { <CustomComponentPropsI32 props = 0 props = 1 /> };
   |                                                ^^^^^

error: Duplicate attribute `props`
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:15:60
   |
15 |     html! { <CustomComponentPropsI32 props = 0 key = "key" props = 1 /> };
   |                                                            ^^^^^

error: Duplicate attribute `key`
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:16:51
   |
16 |     html! { <CustomComponentPropsI32 key = "key1" key = "key2" /> };
   |                                                   ^^^

error: Duplicate attribute `key`
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:17:61
   |
17 |     html! { <CustomComponentPropsI32 key = "key1" props = 0 key = "key2" /> };
   |                                                             ^^^

error: expected `=`
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:18:44
   |
18 |     html! { <CustomComponentPropsI32 props /> };
   |                                            ^

error: Expected a literal or an expression block
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:19:46
   |
19 |     html! { <CustomComponentPropsI32 props = /> };
   |                                              ^

error: Expected a literal or an expression block
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:20:46
   |
20 |     html! { <CustomComponentPropsI32 props = key = "key" /> };
   |                                              ^^^

error: expected `=`
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:21:42
   |
21 |     html! { <CustomComponentPropsI32 key /> };
   |                                          ^

error: Expected a literal or an expression block
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:22:44
   |
22 |     html! { <CustomComponentPropsI32 key = /> };
   |                                            ^

error: Expected a literal or an expression block
  --> tests/html_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:23:44
   |
23 |     html! { <CustomComponentPropsI32 key = props = 1 /> };
   |                                            ^^^^^
